### ISSUE TITLE
running index.php from the cli doesn't allow for symlinks to be used.

### ISSUE THREAD

**Issue by dmlb2000** [bcit-ci/CodeIgniter#5003]
According to the documentation one should be able to run code igniter on the cli by doing the following `php index.php CONTROLLER METHOD`. However, if you setup your git repository with symlinks and submodules the index.php doesn't seem to follow that file system layout.

The setup is as follows...

 1. add a git submodule for the CI main git repo to your local repository.
 2. symlink index.php to the submodule index.php.
 3. symlink the system directory into the submodule.
 4. manage your own applications and resources directories in the current repository.

Example: https://github.com/EMSL-MSC/pacifica-upload-status

Seems like the php `__FILE__` does dereference symlinks which causes running the index.php on the cli to not find the controller specified.

changing the `chdir(dirname(__FILE__))` in index.php to something like the following would help

```
if (isAbsolutePath($_SERVER['SCRIPT_FILENAME'])) {
    chdir(dirname($_SERVER['SCRIPT_FILENAME']));
} else {
    chdir(dirname(join('/', array(getcwd(), $_SERVER['SCRIPT_FILENAME']))));
}
```
(isAbsolutePath isn't defined here and is an exercise for the reader.)

The other solution would be to somehow include multiple paths to resolve applications from, which maybe a bigger issue. Setting the `$applications` folder to something that maybe an array would be better or could be appended to by environment would also help.

**Comment by narfbg** [comment]
It's not supposed to be used through a symbolic link. Copy it over and do whatever you need with it.

And I have to say, you're also not supposed to run applications directly from git or any other VCS.

**Comment by dmlb2000** [comment]
In case others find this I probably should document the fix we came up with....

Yes, removing the .git directory is part of our deployment though after some research I figured out how to use the `git archive` command and performed the following (roughly).

First, `git archive` the CI repository excluding the application directory and unpack the archive into a temporary directory. Then `git archive` your application and resources directories  and unpack the archive in the same temporary directory and over top of the CI code. This got us a clean enough deployment tree that we could pack up and distribute to production systems.
